core/client/fs/src/main/java/alluxio/client/file/FileSystemContext.java

```diff
@@ -259,14 +259,14 @@ private synchronized void closeContext() throws IOException {
       // developers should first mark their resources as closed prior to any exceptions being
       // thrown.
       mClosed.set(true);
-      mWorkerGroup.shutdownGracefully(1L, 10L, TimeUnit.SECONDS);
       mFileSystemMasterClientPool.close();
       mFileSystemMasterClientPool = null;
       mBlockMasterClientPool.close();
       mBlockMasterClientPool = null;
       for (BlockWorkerClientPool pool : mBlockWorkerClientPool.values()) {
         pool.close();
       }
+      mWorkerGroup.shutdownGracefully(1L, 10L, TimeUnit.SECONDS);
```

Comment:
Can you add a comment in the code of what the order should be? (probably a simple comment similar to the PR description?)